nullaway/src/main/java/com/uber/nullaway/GenericsCheck.java

```diff
@@ -0,0 +1,145 @@
+package com.uber.nullaway;
+
+import com.google.common.collect.ImmutableSet;
+import com.google.errorprone.VisitorState;
+import com.google.errorprone.util.ASTHelpers;
+import com.sun.source.tree.ParameterizedTypeTree;
+import com.sun.source.tree.Tree;
+import com.sun.tools.javac.code.Attribute;
+import com.sun.tools.javac.code.Type;
+import com.sun.tools.javac.tree.JCTree;
+import java.util.HashSet;
+import java.util.List;
+
+public class GenericsCheck {
+
+  // check that type is a valid instantiation of its generic type
+  public static void checkInstantiatedType(
+      Type type, VisitorState state, Tree tree, NullAway currentState, Config config) {
+    CodeAnnotationInfo codeAnnotationInfo = CodeAnnotationInfo.instance(state.context);
+    if (!config.isJSpecifyMode()) {
+      throw new IllegalStateException("should only check instantiated types in JSpecify mode");
+    }
```

Comment:
I wonder if rather than forcing the caller to check `isJSpecifyMode()` before calling this method, we should just check it here and bail out if it's not set.  What do you think?